cas/cas.go

```diff
@@ -405,6 +486,82 @@ nextKey:
 	return "", fmt.Errorf("aci not found")
 }
 
+func (ds Store) RemoveTreeStore(key string) error {
+	key, err := ds.ResolveKey(key)
+	if err != nil {
+		return fmt.Errorf("error resolving key: %v", err)
+	}
+	treeStoreKeyLock, err := lock.ExclusiveKeyLock(ds.treeStoreLockDir, key)
+	if err != nil {
+		return fmt.Errorf("error locking tree store: %v", err)
+	}
+	defer treeStoreKeyLock.Close()
+
+	err = ds.treestore.Remove(key)
+	if err != nil {
+		return fmt.Errorf("error removing the tree store: %v", err)
+	}
+	return nil
+}
+
+func (ds Store) RemoveACI(key string) error {
+	key, err := ds.ResolveKey(key)
+	if err != nil {
+		return fmt.Errorf("error resolving key: %v", err)
+	}
+	imageKeyLock, err := lock.ExclusiveKeyLock(ds.imageLockDir, key)
+	if err != nil {
+		return fmt.Errorf("error locking image: %v", err)
+	}
+	defer imageKeyLock.Close()
+
+	// Firstly remove aciinfo and remote from the db in an unique transaction.
+	// remote needs to be removed or a GetRemote will return a blobKey not
+	// referenced by any ACIInfo.
+	err = ds.db.Do(func(tx *sql.Tx) error {
+		found := false
+		_, found, err := GetACIInfoWithBlobKey(tx, key)
```

Comment:
You could place it inside an `if`:
```
if _, found, err := Get...(tx,key); err != nil {
	...
} else if !found {
	...
}
```
That limits the scope of found to `if`, `else` only.